test/unit/domain/consents/revoke.test.ts

```diff
@@ -0,0 +1,229 @@
+  },
+  params: {
+    id: '1234'
+  },
+  payload: {
+    id: '1234',
+    requestId: '475234',
+    initiatorId: 'pispa',
+    participantId: 'sfsfdf23',
+    scopes: [
+      {
+        accountId: '3423',
+        actions: ['acc.getMoney', 'acc.sendMoney']
+      },
+      {
+        accountId: '232345',
+        actions: ['acc.accessSaving']
+      }
+    ],
+    credential: null
+  }
+}
+
+// @ts-ignore
+const requestNoHeaders: Request = {
+  params: {
+    id: '1234'
+  }
+}
+
+/*
+ * Mock Consent Resources
+ */
+const partialConsentActive: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  status: 'ACTIVE'
+}
+
+const partialConsentActive2: Consent = {
+  id: '1234',
+  initiatorId: 'pi2-2233',
+  participantId: 'dfs333-2123',
+  status: 'ACTIVE'
+}
+
+const partialConsentRevoked: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  revokedAt: 'now',
+  status: 'REVOKED'
+}
+
+const completeConsentRevoked: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  status: 'REVOKED',
+  revokedAt: 'now',
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs='
+}
+
+const completeConsentActive: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  credentialId: '123',
+  credentialType: 'FIDO',
+  status: 'ACTIVE',
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs='
+}
+
+const requestBody: SDKStandardComponents.PatchConsentsRequest = {
+  status: 'REVOKED',
+  revokedAt: 'now'
+
+}
+
+describe('server/domain/consents/revoke', (): void => {
+  beforeAll((): void => {
+    mockConsentUpdate.mockResolvedValue(2)
+    mockLoggerError.mockReturnValue(null)
+    mockLoggerPush.mockReturnValue(null)
+  })
+
+  beforeEach((): void => {
+    jest.clearAllMocks()
+  })
+
+  describe('isConsentRequestInitiatedByValidSource', (): void => {
+    it('Should return true', (): void => {
+      expect(
+        isConsentRequestInitiatedByValidSource(partialConsentActive, request))
+        .toBe(true)
+    })
+
+    it('Should return false because consent is null', (): void => {
+      expect(isConsentRequestInitiatedByValidSource(
+        null as unknown as Consent, request))
+        .toBeFalsy()
+    })
+
+    it('Should return false because initiator ID does not match', (): void => {
+      expect(
+        isConsentRequestInitiatedByValidSource(partialConsentActive2, request))
+        .toBeFalsy()
+    })
+
+    it('Should throw an error as request headers are missing', (): void => {
+      expect((): void => {
+        isConsentRequestInitiatedByValidSource(
+          partialConsentActive, requestNoHeaders as Request)
+      }).toThrowError()
+    })
+  })
+
+  describe('revokeConsentStatus', (): void => {
+    it('Should return a revoked consent', async (): Promise<void> => {
```

Comment:
if there is a need to have separate test cases for `partialConsent` and `completeConsent` there should be some comment here why. Assertions are the same and difference is only exposed in parameter to `revokeConsentStatus` and describe description isn't helping to understand. Improve readability, please